docs/custom_loading.rst

```diff
@@ -94,6 +95,10 @@ ensure that the data file being loaded is compatible with the loader function.
         return Spectrum1D(flux=data, wcs=wcs, uncertainty=uncertainty, meta=meta)
 
 
+An ``extensions`` keyword can be provided. This allows for basic filename
```

Comment:
This is for *creating* the extension part, but what about accessing it?  That is, if user code wants to ask things like "what are all the loaders for extension `fits`", that needs to be possible (and documented at least briefly here).